### ISSUE TITLE
Allow attributes to be excluded from SA1117 analysis

### ISSUE THREAD

**Issue by bordecal** [DotNetAnalyzers/StyleCopAnalyzers#1712]
StyleCop did not include attributes when evaluating SA1117 (ParametersMustBeOnSameLineOrSeparateLines).  This allowed us to do some convenient things like

```C#
[SuppressMessage("Foo", "SomeRule",
    Justification = "...")]
```
which is particularly handy when the justification is going to be long-ish.  Obviously, other attributes with a potentially long text argument might also benefit.

A couple of potential ways to address this:
1. Completely ignore attributes for backward compatibility
2. Enable configuration of a list of attribute types for which the rule should be ignored

Personally, I kind of like #2, but it's obviously more work, and we've been living with #1 for a long time now...

**Comment by sharwell** [comment]
Personally I would prefer one of the following:

1. Document our deviation from StyleCop Classic (continue to enforce SA1117 for attributes), and make sure the SA1117 code fix can correct all of these automatically.
1. Combine the previous item with a **stylecop.json** option which disables SA1117's analysis of attributes.

I lean towards the first item here primarily because every provided option is more work to fully test.

**Comment by bordecal** [comment]
Whoops...  I'd started a list of 3 options, and then couldn't remember one of them, which turns out to be your \#2.  That would be fine too, but keeping the current, non-StyleCop compatible behavior without an alternative is not great when it has a non-trivial impact on readability in at least some cases.

(FWIW, if the implementation and testing effort a big concern, I'd be happy to contribute in lieu of writing my own replacement rule.  However, I can't really afford to spend the time unless there's an agreement around the target approach...)

**Comment by bordecal** [comment]
This caused me enough pain with a brownfield project this week that ended up just going ahead and implementing the change.  I'm not overly fond of the either the default or the ignore options, so I added a third option that I prefer over either of those: allowing the first split (if any) to occur after the positional parameters.  This fits quite well with the way I've seen `SuppressMessage` attributes applied on many projects that require justifications and have line length limitations.
